crates/flashblocks-rpc/src/rpc.rs

```diff
@@ -387,7 +395,23 @@ where
                         block_number: Some(block_number),
                         base_fee,
                     };
-                    Ok(Some(self.transform_tx(transaction, tx_info)))
+                    // preload transaction receipt if it's a deposit transaction
+                    if transaction.is_deposit() {
+                        let receipt =
+                            EthTransactions::transaction_receipt(&self.eth_api, tx_hash).await;
+
+                        // since this txn hash exists in a block, we should always be able to fetch a receipt for it
+                        let envelope: OpReceiptEnvelope = receipt.unwrap().unwrap().into();
```

Comment:
Generally the semantics in Rust, you avoid unwraps whenever possible, you can use matching and allow the calling method to handle the errors.

Reply:
I'm slightly unsure here what we should do if this actually errors. Typically that should never happen given the context we're operating in - but in case our state is messed up for some reason and it does happen, do we just return `None` as the result for this `getTransactionByHash` call? cc @danyalprout as well

Reply:
Agree with Joe, would be great if we could propagate this error back to the client (& error log on our side) vs. panicking

Reply:
https://github.com/base/node-reth/pull/28/commits/c231313c0f86ac03784e36f3d0ea2f98b53a5bd6